druid-shell/src/platform/x11/window.rs

```diff
@@ -314,64 +292,37 @@ impl Window {
         xcb::destroy_window(self.app.connection(), self.id);
     }
 
-    fn cairo_surface(&self) -> Result<XCBSurface, Error> {
-        match self.cairo_context.try_borrow() {
-            Ok(ctx) => match ctx.get_target().try_into() {
-                Ok(surface) => Ok(surface),
-                Err(err) => Err(Error::Generic(format!(
-                    "Window::cairo_surface ctx.try_into: {}",
-                    err
-                ))),
-            },
-            Err(err) => Err(Error::BorrowError(format!(
-                "Window::cairo_surface cairo_context: {}",
-                err
-            ))),
-        }
+    fn cairo_surface(&self) -> Result<XCBSurface, anyhow::Error> {
+        self.cairo_context
+            .try_borrow()
+            .context("Window::cairo_surface cairo_context")?
+            .get_target()
+            .try_into()
+            .map_err(|_| anyhow!("Window::cairo_surface try_into"))
```

Comment:
What's the benefit of `map_err` over `context`? Especially as we're dropping the underlying error.

Reply:
The underlying "error" here is just cairo returning the surface back to us, so it doesn't contain any useful information.